protocol/op-deployer-embedded-artifacts.md

```diff
@@ -0,0 +1,67 @@
+will reduce the complexity of our deployments, and prevent OP Deployer and the contract artifacts from becoming out
+of sync.
+
+## Problem Statement + Context
+
+Today, OP Deployer lets users provide an "artifacts locator" which tells the tool where to find the contract
+artifacts to use for deployment. The locator can be a `file://` url, an `https://` url, or a `tag://` URL. File and
+HTTPs locators are only used in dev for special cases like unit tests, and therefore are not a concern here. Tagged
+locators, however, are used for production deployments and are a huge area of confusion for users for the following
+reasons:
+
+- The artifacts that a tagged locator points to are created from a different branch than OP Deployer itself. This
+  can lead to a mismatch between OP Deployer's deployment code and the artifacts it is trying to deploy.
+- There is a 1:1 mapping between each OP Deployer version and the tagged versions it supports. `v0.2.0` of OP
+  Deployer, for example, only supports smart contracts from `op-contracts/v2.0.0`. However, the version of the smart
+  contracts can sometimes change by minor or patch versions. This makes it difficult to determine which version of
+  the contracts each version of OP Deployer supports, and can lead to unexpected behavior or broken deployments.
+
+In addition, maintaining the tagged locators themselves is non-trivial amount of work:
+
+- New releases require contract artifacts to be built, pushed to GCP, and added to `standard.go`. This means that
+  every new contract release requires a corresponding code change to OP Deployer.
+- Because the artifacts each locator points to are on GCP, they need to be validated client-side before being used.
+  This adds complexity, overhead, and potential for security vulnerabilities.
+- It's much less clear which commit a given set of artifacts belong to. Artifacts are built using a
+  content-addressable hash of their source code. A commit that doesn't change the source code won't change the hash,
+  so the commit for a given contract release may not match the commit in the uploaded artifacts. While not a huge
+  issue, this makes determining the provenance of a given set of artifacts quite difficult.
+
+## Proposed Solution
+
+Rather than continuing to support tagged locators, we propose embedding the contract artifacts for a given release
+directly inside the OP Deployer binary itself using `go:embed`. HTTPs and file locators will continue to work as
+before. The CLI will select the tagged locator by default if it exists, so users will not generally have to specify
+a locator at all when invoking OP Deployer.
+
+This change would increase the size of the OP Deployer binary by ~60MB after removing test code from the artifacts.
+It is our belief that this is an acceptable tradeoff, particularly since users would have to download the artifacts
+anyway.
```

Comment:
Does the ~60MB include both contract source code AND forge scripts? I'm assuming we'd want to embed both